include/eve/arch/riscv/tags.hpp

```diff
@@ -0,0 +1,118 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/cardinals.hpp>
+#include <eve/arch/cpu/tags.hpp>
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/concept/scalar.hpp>
+#include <eve/detail/meta.hpp>
+
+#include <climits>
+
+namespace eve
+{
+//================================================================================================
+// ABI tags for all RISCV bits SIMD registers
+//================================================================================================
+template<std::size_t Size> struct rvv_abi_
+{
+  static_assert(CHAR_BIT == 8, "[eve riscv] - For riscv we expect CHAR_BIT to be 8");
+  static constexpr std::size_t MaxLmul         = 8;
+  static constexpr std::size_t bits            = Size * MaxLmul;
+  static constexpr std::size_t bytes           = bits / 8;
+  static constexpr bool        is_wide_logical = false;
+
+  template<typename Type>
+  static constexpr bool is_full = ((Type::size() * sizeof(typename Type::value_type)) >= bytes);
+
+  template<typename Type>
+  static constexpr std::size_t fundamental_cardinal = Size / 8 / sizeof(Type);
```

Comment:
On a surface this does not strike me as correct.

fundamental cardinal is cardinal for which we have a native register. Like xmm on avx2.

In my mind we differ in terminology with RISK-V here - everything beyond max length requires masking and isn't fundamental cardinal.

Reply:
Well, I thought about fundamental cardinal as the 
```cpp
the smallest register able to store values of type of type `Type` for a given SIMD `ABI` with no uninitialized lanes
```
(in file include/eve/arch/fundamental_cardinal.hpp)

So in this implementation fundamental cardinal is one vector register (actually, for some cases we can use fractional LMUL) and expected cardinal is the maximum available vector registers, combined together.